root/tree/tree/h5hep/h5hep/common.hxx

```diff
@@ -0,0 +1,140 @@
+/// \file h5hep/common.hxx
+/// \author Javier Lopez-Gomez <j.lopez@cern.ch>
+/// \date 2021-10-23
+///
+/// \brief Header-only library to read/write high-energy physics data in HDF5.
+/// For benchmark purposes only; do NOT use in production.
+
+#ifndef _H5HEP_COMMON_HXX
+#define _H5HEP_COMMON_HXX
+
+#include <hdf5.h>
+
+#include <cassert>
+#include <type_traits>
+#include <vector>
+
+#if defined(__GNUC__)
+# define likely(x)   __builtin_expect(!!(x), 1)
+# define unlikely(x) __builtin_expect(!!(x), 0)
+#else
+# define likely(x)   x
+# define unlikely(x) x
+# define __restrict
```

Comment:
Maybe warn here that we don't support these if they are important for optimization.